langchain4j-azure-open-ai/src/main/java/dev/langchain4j/model/azure/AzureOpenAiStreamingChatModel.java

```diff
@@ -93,12 +96,20 @@ public AzureOpenAiStreamingChatModel(OpenAIClient client,
                                          List<AzureChatExtensionConfiguration> dataSources,
                                          AzureChatEnhancementConfiguration enhancements,
                                          Long seed,
-                                         ChatCompletionsResponseFormat responseFormat) {
+                                         ChatCompletionsResponseFormat responseFormat,
+                                         boolean useAsyncClient) {
 
         this(deploymentName, tokenizer, maxTokens, temperature, topP, logitBias, user, n, stop, presencePenalty, frequencyPenalty, dataSources, enhancements, seed, responseFormat);
-        this.client = client;
-    }
 
+        if(useAsyncClient) {
```

Comment:
Pinging @langchain4j as this needs to be consistent with the rest of the API:
if the `useAsyncClient` boolean really needed? If `asyncClient` is set, then we'll use it, otherwise we'll switch to the sync client.

Reply:
replaced the logic for useAsyncClient with asyncClient Object from constructor.